File: AbstractFactoryExercise/AbstractFactoryExercise/ConcreteFactories/DungeonFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AbstractFactoryExercise.Abstractions;
using AbstractFactoryExercise.Constants;
using AbstractFactoryExercise.Products;

namespace AbstractFactoryExercise.ConcreteFactories
{
    public class DungeonFactory : Abstractions.LevelFactory
    {
        public override string[] GetAvailableEnemyNames()
        {
            return new string[]
                {
                "Knight",
                "Archer",
                };
        }
        public override Abstractions.Enemy CreateEnemy(string enemyName)
        {
            return enemyName switch
            {
                "Knight" => new Knight(),
                "Archer" => new Archer(),
                _ => throw new ArgumentException(GlobalConstants.INVALID_CHOICE)
            };
        }
        public override Abstractions.Environment CreateEnvironment()
        {
            return new CastleWall();
        }
    }
}

File: AbstractFactoryExampleJava/src/concretefactories/DungeonFactory.java
package concretefactories;

import abstractions.Enemy;
import abstractions.Environment;
import abstractions.LevelFactory;
import constants.GlobalConstants;
import products.Archer;
import products.CastleWall;
import products.Knight;

public class DungeonFactory extends LevelFactory {
    @Override
    public String[] getAvailableEnemyNames() {
        return new String[] { "Knight", "Archer" };
    }

    @Override
    public Enemy createEnemy(String enemyName) {
        return switch (enemyName) {
            case "Knight" -> new Knight();
            case "Archer" -> new Archer();
            default -> throw new IllegalArgumentException(GlobalConstants.INVALID_CHOICE);
        };
    }

    @Override
    public Environment createEnvironment() {
        return new CastleWall();
    }
}